app/services/throttle.rb

```diff
@@ -108,8 +103,7 @@ def fetch_state!
       @redis_attempted_at = nil
     else
       @redis_attempted_at =
-        Time.zone.now -
-        Throttle.attempt_window_in_minutes(throttle_type).minutes + ttl.seconds
+        Time.zone.at(expiretime).utc - Throttle.attempt_window_in_minutes(throttle_type).minutes
```

Comment:
`Time.zone` will already be UTC in most cases (except for zonebie in specs), so the `#utc` will be duplicative?
Maybe we just hardcode UTC?
```suggestion
        ActiveSupport::TimeZone['UTC']at(expiretime) - Throttle.attempt_window_in_minutes(throttle_type).minutes
```